ultra/ultra/train.py

```diff
@@ -115,6 +122,76 @@ def create_argument_parser():
     return parser
 
 
+def load_model(experiment_dir):
+    # Load relevant agent metadata.
+    with open(
+        os.path.join(experiment_dir, "agent_metadata.pkl"), "rb"
+    ) as metadata_file:
+        agent_metadata = pickle.load(metadata_file)
+
+    for key in agent_metadata.keys():
+        assert key in ["agent_ids", "agent_classes", "agent_specs"]
+
+    # Extract the agent IDs and policy classes from the metadata and given models.
+    agent_ids = [agent_id for agent_id in agent_metadata["agent_ids"]]
+
+    agent_classes = {
+        agent_id: agent_metadata["agent_classes"][agent_id] for agent_id in agent_ids
+    }
+
+    agent_checkpoint_directories = {
+        agent_id: sorted(
+            glob.glob(os.path.join(experiment_dir, "models", agent_id, "*")),
+            key=lambda x: int(x.split("/")[-1]),
+        )
+        for agent_id in agent_ids
+    }
+
+    # Confined to single agent experimens only
+    assert len(agent_ids) == 1, "Cannot load model from multi-agent experiments."
+
+    length_dir = len(agent_checkpoint_directories[agent_ids[0]])
+    checkpoint_directory = agent_checkpoint_directories[agent_ids[0]][length_dir - 1]
+    if length_dir > 1:
+        print(
+            f"\nThere are {length_dir} models inside in the experiment dir. Only the latest model >>> {checkpoint_directory} <<< will be trained\n"
+        )
+
+    # Create the agent specifications matched with their associated ID and corresponding
+    # checkpoint directory
+    agent_specs = {
+        agent_id: make(
+            locator=agent_classes[agent_id],
+            checkpoint_dir=checkpoint_directory,
+            experiment_dir=experiment_dir,
+            agent_id=agent_id,
+        )
+        for agent_id in agent_ids
+    }
+
+    # Create the agents matched with their associated ID.
+    agents = {
+        agent_id: agent_spec.build_agent()
+        for agent_id, agent_spec in agent_specs.items()
+    }
+
+    try:
+        _ = [
+            agents[agent_id].load_replay_buffer(
+                os.path.join(experiment_dir, "replay_buffers", agent_id)
+            )
+            for agent_id in agent_ids
+        ]
```

Comment:
Will this work for agents where some have a replay buffer and some do not have a replay buffer?

For example, say the `agent_ids` are `["000", "001"]` where 000 does not have a replay buffer, but 001 does have a replay buffer. As we iterate through the IDs, the `NotImplementedError` will be thrown on the first iteration since 000 does not have a replay buffer, and we will be immediately redirected to the `except` block without 001 ever having loaded its replay buffer.

Is this a possibility?

Reply:
Will do 👍